text/001x-battery_improvements.md

```diff
@@ -0,0 +1,110 @@
+      <field type="int32_t" name="current_consumed" units="mAh" invalid="-1">Consumed charge, -1: Current consumption estimate not provided.</field>
+      <field type="int32_t" name="energy_consumed" units="hJ" invalid="-1">Consumed energy, -1: Energy consumption estimate not provided.</field>
+      <field type="int8_t" name="battery_remaining" units="%" invalid="-1">Remaining battery energy. Values: [0-100], -1: Remaining battery energy is not provided.</field>
+      <field type="uint32_t" name="time_remaining" units="s" invalid="UINT32_MAX">Remaining battery time (estimated), UINT32_MAX: Remaining battery time estimate not provided.</field>
+      <field type="uint8_t" name="charge_state" enum="MAV_BATTERY_CHARGE_STATE">State for extent of discharge, provided by autopilot for warning or external reactions</field>
+      <field type="uint8_t" name="mode" enum="MAV_BATTERY_MODE">Battery mode. Default (0) is that battery mode reporting is not supported or battery is in normal-use mode.</field>
+      <field type="uint32_t" name="fault_bitmask" display="bitmask" enum="MAV_BATTERY_FAULT">Fault/health indications. These should be set when charge_state is MAV_BATTERY_CHARGE_STATE_FAILED or MAV_BATTERY_CHARGE_STATE_UNHEALTHY (if not, fault reporting is not supported).</field>
+    </message>
+```
+
+
+Questions:
+- It is desirable to move static fields like the `type` (battery chemistry) and `battery_function` out of the new message because they only need to be read once, and afterwards are dead weight.
+  For this to be OK, the information either has to be non-essential (i.e might never be provided) or guaranteed to be sent in another message (this info is present in [SMART_BATTERY_INFO](https://mavlink.io/en/messages/common.html#SMART_BATTERY_INFO)).
+  - Is it non-essential?
+  - If it is essential, do we leave it in the message or mandate sending of `SMART_BATTERY_INFO`?
+- Do we need all the other fields?
+- Are there any other fields missing?
+
+
+## Battery Voltages
+
+The proposed battery message is below.
+
+This assumes that the reason you might need the individual battery voltages is in order to identify that a particular cell has a significantly different voltage, and is hence in fault.
+It simply sends information about which cells are in fault, providing battery manufacturers a way of reporting more detailed debugging information.
+
+```xml
+    <message id="???" name="BATTERY_VOLTAGE_FAULT">
+      <description>Battery information. Updates GCS with flight controller battery status. Smart batteries also use this message, but may additionally send SMART_BATTERY_INFO.</description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
+      <field type="uint8_t" name="index">Cell index (0 by default). This is iterated for every 64 cells, allowing very large cell fault information. </field>
+      <field type="uint64_t" name="fault_mask">Fault/health indications. .</field>
+    </message>
```

Comment:
If this messages corresponds to a single cell, why don't we add an extra field for the cell voltage value itself?

Reply:
@dakejahl It corresponds to a single battery - so this gives you **fault** information on every cell in the battery. If there are more than 64 cells, you send it again with an increased cell index value.

Yes we could include the total battery voltage, but you get that in battery status. If you're sending fault information it isn't useful/and isn't actually relevant.

I've had feedback that this message is not suitable. The argument behind this design was that you are only interested in the cells that are in fault, and it would be the smart battery that told you. In other words that you do not need to know the actual voltages for the cells.

However I have since been told that when your cells start to go it is actually likely that a human will be the right person to work that out/analyse the data. So instead of sending faults we should send the raw cell data.
note however that you don't need it all the time - just when there is a problem. 
I will modify in line with that later today.

Reply:
Note, the message changed to `BATTERY_CELL_VOLTAGES`. The answer still applies, so I am closing this a resolved.